internal/httpginendpoints/httpginendpoints.go

```diff
@@ -111,13 +112,115 @@ func PostRawAttest(c *gin.Context) {
 
 	reportData := attest.GenerateMAAReportData(runtimeDataBytes)
 	rawReport, err := attestationReportFetcher.FetchAttestationReportHex(reportData)
+	// TODO: review this StatusForbidden - surely should be StatusInternalServerError
 	if err != nil {
 		c.JSON(http.StatusForbidden, gin.H{"error": err.Error()})
 	}
 
 	c.JSON(http.StatusOK, gin.H{"report": rawReport})
 }
 
+/*
+	As PostRawAttest but also the various certificates and the UVM reference info so as to suit the Privacy Sandbox.
+
+	see https://github.com/microsoft/azure-privacy-sandbox-kms/blob/main/test/attestation-samples/snp.json
+	{
+	"endorsed_tcb": "0300000000000873",
+	"endorsements": "base64 encoded certificate chain",
+	"evidence": "base64 encoded attestation report",
+	"uvm_endorsements": "base64 encoded uvm reference info COSESign1 document",
+	}
+
+	This is mostly easily obtainable from /security-context-* but this endpoint is provided for convenience.
+	It may also choose to do a better job of fetching the AMD erts, eg othger sources than the local THIM and with
+	better retries.
+*/
+
+type CombinedAttestationData struct {
+	// PSP TCB version
+	Endorsed_tcb string `json:"endorsed_tcb"`
+	// AMD certificate chain matching the attestation report
+	Endorsements string `json:"endorsements"`
+	// attestation report base64 encoded
+	Evidence string `json:"evidence"`
+	// In the absence of managed identity assignment to the container group
+	// an AAD token issued for authentication with AKV resource may be included
+	// in the request to release the key.
+	Uvm_endorsements string `json:"uvm_endorsements"`
+}
+
+func PostCombinedAttest(c *gin.Context) {
+	var attestData RawAttestData
+
+	logrus.Debug("PostCombinedAttest...")
+	// Call BindJSON to bind the received JSON to AttestData
+	if err := c.ShouldBindJSON(&attestData); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": errors.Wrapf(err, "invalid request format").Error()})
+		return
+	}
+
+	uvmInfo, ok := c.MustGet("uvmInfo").(*common.UvmInformation)
+	if !ok {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errors.New("uvmInfo is not set")})
+		return
+	}
+
+	// base64 decode the incoming encoded security policy
+	inittimeDataBytes, err := base64.StdEncoding.DecodeString(uvmInfo.EncodedSecurityPolicy)
+
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errors.Wrap(err, "decoding policy from Base64 format failed").Error()})
+		return
+	}
+
+	// standard base64 decode the incoming runtime data
+	runtimeDataBytes, err := base64.StdEncoding.DecodeString(attestData.RuntimeData)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": errors.Wrapf(err, "decoding base64-encoded runtime data of request failed").Error()})
+		return
+	}
+
+	var attestationReportFetcher attest.AttestationReportFetcher
+	if attest.IsSNPVM() {
+
```

Comment:
Nit: Is there a reason for this large space?

Reply:
I would do a logrus saying that we are an SNP VM and using the attestation report fetch